py/desitarget/mock/build.py

```diff
@@ -1042,3 +1060,509 @@ def join_targets_truth(mockdir, outdir=None, force=False, comm=None):
         mtl.meta['EXTNAME'] = 'MTL'
         mtl.write(tmpout, overwrite=True, format='fits')
         os.rename(tmpout, out_mtl)
+
+def _get_magnitudes_onebrick(specargs):
+    """Filler function for the multiprocessing."""
+    return get_magnitudes_onebrick(*specargs)
+
+def get_magnitudes_onebrick(target_name, mockformat, thisbrick, brick_info, Magnitudes, dust_dir,
+                         select_targets_function, source_data, rand, log):
+    """Wrapper function to generate magnitudes for all the objects on a single brick."""
+
```

Comment:
I agree we shouldn't be replicating tons of code. Once all the big current and future issues are solved in the full spectra version I will refactor the `--no-spectra` option into the latest version. 

Admittedly, this is a quick hack to avoid the previous state where we lost the option of running fast catalog only end-to-end simulations because of major ongoing work in `select_targets_truth`.